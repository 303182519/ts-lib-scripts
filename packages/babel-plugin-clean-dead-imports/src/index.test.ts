import { transformAsync } from '@babel/core';
import { join } from 'path';
import plugin from './index';

it('移除无用导入', async () => {
  const input = `
    import A, { B, C, D, E } from './E';

    export { B, /* type */ C };
  `;

  const result = await transformAsync(input, {
    filename: 'test.ts',
    babelrc: false,
    cwd: join(__dirname, '.'),
    presets: [
      [
        '@babel/preset-env',
        {
          modules: false,
        },
      ],
    ],
    plugins: [plugin],
  });

  expect(result?.code).toMatchSnapshot();
});
