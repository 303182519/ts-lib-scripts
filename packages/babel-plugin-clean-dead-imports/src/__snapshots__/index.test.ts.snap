// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`移除无用导入 1`] = `
"import A, { B } from './E';
export { B };"
`;
